### ISSUE TITLE
Typescript knex.fn.now() update support

### ISSUE THREAD

**Issue by maximelkin** [knex/knex#3652]
# Environment

Knex version: latest
Database + version: any
OS: any

@lorefnon.

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do

```typescript
interface User {
    updatedAt: Date
}

// error at updatedAt: ...
await knex<User>('users').update({ updatedAt: knex.fn.now() });
```

2. Error message

TS2345: Argument of type '{ updatedAt: Knex.Raw<Date>; }' is not assignable to parameter of type 'MaybeArray<Partial<User>>'.   Type '{ updatedAt: Knex.Raw<Date>; }' is not assignable to type 'Partial<User>'.     Types of property 'updatedAt' are incompatible.       Type 'Raw<Date>' is missing the following properties from type 'Date': toDateString, toTimeString, toLocaleDateString, toLocaleTimeString, and 37 more.

**Comment by KristjanTammekivi** [comment]
IIRC knex.fn.now() essentially returns knex.raw(). IMO Knex should allow raw for every select/update field. Should be easy enough to do, might take this on myself

**Comment by kibertoad** [comment]
Released in  0.20.14.
